### PR TITLE
[#153] Refactoring create index

### LINKED ISSUE #153: Remove parameter `field` in database.create_field_index()

### PR DESCRIPTION
**Pull request by eddiebrissow**
closes #153 
Removed 'field' from create index method

### CODE REVIEW

File: hyperon_das_atomdb/index.py [hunk 1]
```diff
@@ -18,14 +18,21 @@ def generate_index_id(field: str, conditionals: Dict[str, Any]) -> str:
...
 
         Args:
-            field (str): The field to create the index on.
+            atom_type (str): Atom's type
+            fields (List[str]): The fields to create the index on.
+
 
         Returns:
-            Tuple[str, Any]: The index ID.
+            Tuple[str, Any]: Returns the index id and the index proprieties dict
```
Comment (andre-senna) [inline hyperon_das_atomdb/index.py:35]: Typo: "properties"

**Review by andre-senna — APPROVED** [review]

File: hyperon_das_atomdb/adapters/redis_mongo_db.py [hunk 2]
```diff
@@ -108,36 +108,28 @@ def create(
...
         if index_type == MongoIndexType.TEXT:
-            index_list = [(f, 'text') for f in fields] if fields is not None else [(field, 'text')]
+            index_list = [(f, 'text') for f in fields]
         else:
-            index_list = (
-                [(f, ASCENDING) for f in fields]
-                if fields is not None
-                else [(field, ASCENDING)]  # store the index in ascending order
-            )
+            index_list = [(f, ASCENDING) for f in fields]  # store the index in ascending order
```
Comment (angeloprobst) [inline hyperon_das_atomdb/adapters/redis_mongo_db.py:132]: What happens here if `fields` as passed in as an empty list?

File: hyperon_das_atomdb/adapters/redis_mongo_db.py [hunk 3]
```diff
@@ -995,18 +987,14 @@ def delete_atom(self, handle: str, **kwargs) -> None:
...
         type: Optional[str] = None,
         composite_type: Optional[List[Any]] = None,
-        fields: Optional[List[str]] = None,
         index_type: Optional[FieldIndexType] = None,
     ) -> str:
         if type and composite_type:
             raise ValueError("Both type and composite_type cannot be specified")
 
-        if field is None and fields is None:
-            raise ValueError("No field to create index: field and fields can not be None")
```
Comment (angeloprobst) [inline hyperon_das_atomdb/adapters/redis_mongo_db.py:1008]: Same here. Shouldn't `fields` be checked to ensure it's not empty?

File: hyperon_das_atomdb/adapters/ram_only.py [hunk 4]
```diff
@@ -538,10 +538,9 @@ def delete_atom(self, handle: str, **kwargs) -> None:
     def create_field_index(
         self,
         atom_type: str,
-        field: str,
+        fields: List[str],
         type: Optional[str] = None,
```
Comment (angeloprobst) [inline hyperon_das_atomdb/adapters/ram_only.py:542]: `type` is a builtin function in python, so, ideally, it should not be used to name vars/params.

File: hyperon_das_atomdb/adapters/redis_mongo_db.py (hunk 2 above)
Comment (eddiebrissow) [reply hyperon_das_atomdb/adapters/redis_mongo_db.py:134]: Added verification

File: hyperon_das_atomdb/adapters/redis_mongo_db.py (hunk 3 above)
Comment (eddiebrissow) [reply hyperon_das_atomdb/adapters/redis_mongo_db.py:1008]: Added verification

File: hyperon_das_atomdb/adapters/ram_only.py (hunk 4 above)
Comment (eddiebrissow) [reply hyperon_das_atomdb/adapters/ram_only.py]: fixed

**Review by angeloprobst — APPROVED** [review]

**Pull request merged**